.github/workflows/test-go-task.yml

```diff
@@ -197,6 +197,7 @@ jobs:
       - name: Send unit tests coverage to Codecov
         uses: codecov/codecov-action@v3
         with:
+          token: db21daa9-939e-4742-be68-a9db33b8bdcb
```

Comment:
```suggestion
          token: ${{ vars.CODECOV_TOKEN }}
```

I gave it more thought and I don't think it is a good idea to hardcode into the workflow in this way after all.

I'm not too concerned about malicious use of the token because an attacker can't do anything of value with it. However, I think there is the potential for inadvertent harm by legitimate contributors and other community members. With the token hardcoded into the workflow, the coverage data from runs in forks will be uploaded to Arduino's codecov account. That isn't much of a problem if the run is in a feature branch, but it is fairly common for contributors to stage PRs in the default branch, and also any development work in forks would also likely be done in the default branch. This will cause irrelevant data to be added to Arduino's Codecov account.

For example, here is some data that was uploaded from a run triggered by a change I made in the `master` branch in my fork:

![image](https://user-images.githubusercontent.com/8572152/228212673-6f86e167-5092-4da4-837b-ea29cd592678.png)

There is a recently added "repository variables" feature that allows variables to be defined in the repository settings. These are not secrets, so they are available to workflow runs triggered by a PR from a fork:

https://docs.github.com/en/actions/learn-github-actions/variables#defining-configuration-variables-for-multiple-workflows

The variables are not transferred when someone forks a repository, so the token will only be defined when the workflow runs in this repository (the forker could always define their own variable with the token from their Codecov account if they want their own coverage data).

Reply:
> With the token hardcoded into the workflow, the coverage data from runs in forks will be uploaded to Arduino's codecov account.

That was a case I'd liked to try before merging, but you beat me, thanks @per1234 :-)

I'll regenerate the token and convert it to a repository variable.

Reply:
>I'll regenerate the token and convert it to a repository variable.

To accompany my review suggestion, I already added a variable for the existing token value:

![image](https://user-images.githubusercontent.com/8572152/228225044-ba1081cd-fd23-4e95-9258-d0918b2d8d7f.png)


But you're welcome to adjust that however you like, on this page:

https://github.com/arduino/arduino-cli/settings/variables/actions

Note that GitHub wrote some incorrect information on that page:

>Anyone with collaborator access to this repository can use these secrets and variables for actions. They are not passed to workflows that are triggered by a pull request from a fork.

This is correct for secrets, but incorrect for variables. Variables are passed to workflows triggered by PRs from forks.

Reply:
I've updated the token since the old one was published.